### PR TITLE
04 - Model Evaluation

### PR DESCRIPTION
**Pull request by luisquintanilla**

### CODE REVIEW

File: machine-learning/04-Model Evaluation.ipynb [hunk 1]
```diff
@@ -4,25 +4,478 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### Split the data into train and test sets\n",
+        "\n",
+        "The train set is what you'll use to learn the patterns of your data. The test set is the dataset used to evaluate the performance of your model using evaluation metrics for the regression task. \n",
+        "\n",
+        "In this case, 80% of the data is used for training and 20% is used for test/evaluation as defined by the `testFraction` parameter."
```
Comment (beccamc) [inline machine-learning/04-Model Evaluation.ipynb:236]: I would also include what a validation set is used for

File: machine-learning/04-Model Evaluation.ipynb [hunk 2]
```diff
@@ -4,25 +4,478 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### Configure experiment\n",
+        "\n",
+        "Use AutoML to configure our experiment to train for 60 seconds using the pipeline you've just defined. \n",
+        "\n",
+        "By default, AutoML evaluates the models it trains using the evaluation metric you want to optimize. In this case it's R-Squared which is calculated by comaring the actual value `fare_amount` against the predicted value `Score`. "
```
Comment (beccamc) [inline machine-learning/04-Model Evaluation.ipynb:308]: I was interested in hearing more examples here on other metrics I may want to optimize on. Not necessary if you can't think of another good example here.

File: machine-learning/04-Model Evaluation.ipynb [hunk 3]
```diff
@@ -4,25 +4,478 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## What is model evaluation?\n",
+        "\n",
+        "Training is the process of applying algorithms to historical data in order to create a model that accurately represents that data. That model is then used on new data to make predictions. \n",
+        "\n",
```
Comment (JakeRadMSFT) [inline machine-learning/04-Model Evaluation.ipynb:28]:
```suggestion
        "\n",
        "In this notebook you will learn:\n",
        "\n",
        "- What is model evaluation?\n",
        "- What are common evaluation metrics?\n",
        "- How to evaluate a model in ML.NET\n",
        "- Train and evaluate models with cross-validation\n",
        "- Model explainability\n",
        "- How to improve your models - Some ChangeS!"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## What is model evaluation?\n",
        "\n",
        "Training is the process of applying algorithms to historical data in order to create a model that accurately represents that data. That model is then used on new data to make predictions. \n",
        "\n",

```

File: machine-learning/04-Model Evaluation.ipynb [hunk 4]
```diff
@@ -4,25 +4,549 @@
...
+      ],
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### Create training pipeline\n",
+        "\n",
+        "For ths dataset, the following transforms are applie:\n",
```
Comment (beccamc) [inline machine-learning/04-Model Evaluation.ipynb:274]:
```suggestion
        "For ths dataset, the following transforms are applied:\n",

```

File: machine-learning/04-Model Evaluation.ipynb [hunk 5]
```diff
@@ -4,25 +4,549 @@
...
+        "\n",
+        "- OneHotEncoding to convert categorical values into numerical values\n",
+        "- ReplaceMIssingValues which as the name suggests is to replace any missing values.\n",
```
Comment (beccamc) [inline machine-learning/04-Model Evaluation.ipynb:277]:
```suggestion
        "- ReplaceMissingValues which as the name suggests is to replace any missing values.\n",

```

File: machine-learning/04-Model Evaluation.ipynb [hunk 6]
```diff
@@ -4,25 +4,549 @@
...
+          "execution_count": 1,
+          "metadata": {}
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "Once you have an `IDataView` with your predictions, use the `Evaluate` method for the regression task to calculate the evaluation metrics for the test set where the `fare_amount` column is the actual value and the `Score` column is te predicted value."
```
Comment (beccamc) [inline machine-learning/04-Model Evaluation.ipynb:428]:
```suggestion
        "Once you have an `IDataView` with your predictions, use the `Evaluate` method for the regression task to calculate the evaluation metrics for the test set where the `fare_amount` column is the actual value and the `Score` column is the predicted value."

```

**Comment by luisquintanilla** [comment]
Code complete. Need to add md and update based on code review.

File: machine-learning/04-Model Evaluation.ipynb (hunk 1 above)
Comment (luisquintanilla) [reply machine-learning/04-Model Evaluation.ipynb]: resolved

File: machine-learning/04-Model Evaluation.ipynb (hunk 4 above)
Comment (luisquintanilla) [reply machine-learning/04-Model Evaluation.ipynb:286]: Resolved

File: machine-learning/04-Model Evaluation.ipynb (hunk 6 above)
Comment (luisquintanilla) [reply machine-learning/04-Model Evaluation.ipynb]: Resolved

File: machine-learning/04-Model Evaluation.ipynb (hunk 5 above)
Comment (luisquintanilla) [reply machine-learning/04-Model Evaluation.ipynb]: Resolved

File: machine-learning/04-Model Evaluation.ipynb (hunk 2 above)
Comment (luisquintanilla) [reply machine-learning/04-Model Evaluation.ipynb:320]: Resolved.

**Pull request merged**